### DISCUSSION TITLE
Gaussian Blur Support for iOS?

### DISCUSSION THREAD

**Discussion by proxemics** [airbnb/lottie-ios#1701] [Ideas]
With Android recently receiving support for gaussian blur and web generally supporting it as well, was curious if there are any plans or contributions around gaussian blur support for iOS?

Thanks!

**Comment by calda** [comment]
I don't personally plan on working on support for blurs at this time, but we always welcome PRs from the community!

**Reply by calda** [reply]
Does Core Animation natively support rendering gaussian blurs in a `CALayer` on iOS? Lottie on iOS is a hierarchy of `CALayer`s so we wouldn't be able to use the UIKit `UIVisualEffectView` to render dynamic blurs, for example.

**Reply by proxemics** [reply]
Thanks for the reply and insight! I'm just a lowly designer/animator sadly 😅, but I'll see if I can stir up some interest from any Lottie engineers I know.

**Reply by A30008910-wei** [reply]
`CALayer` does support Gaussian Blur with `CIFilter`

```Swift
let layer = CALayer()
     
if let filter = CIFilter(name:"CIGaussianBlur") {
    filter.name = "myFilter"
    layer.backgroundFilters = [filter]
    layer.setValue(1,
                   forKeyPath: "backgroundFilters.myFilter.inputRadius")
}
```

https://developer.apple.com/documentation/quartzcore/calayer/1410827-backgroundfilters

**Reply by calda** [reply]
Does that work on iOS? The docs for that say "This property is not supported on layers in iOS."

**Reply by weisunOW** [reply]
I haven't personally tried it before but the `backgroundFilters` is accessible on iOS. Maybe I will give it a go later in the week.

**Reply by weisunOW** [reply]
If that's not an option, maybe we could try `CAMetalLayer`?

**Reply by DigitalSolomon** [reply]
Has anyone successfully applied a `CIFilter` to a Lottie animation efficiently during playback?
